skimage/_shared/interpolation.pxd

```diff
@@ -169,57 +185,58 @@ cdef inline double biquadratic_interpolation(double* image, Py_ssize_t rows,
                                  r0 + pr, c0 + pc, mode, cval)
         fr[pr] = quadratic_interpolation(xc, fc)
 
-    # cubic interpolation for interpolated values of each row
-    return quadratic_interpolation(xr, fr)
+    out[0] = <np_real_numeric_out>quadratic_interpolation(xr, fr)
 
 
-cdef inline double cubic_interpolation(double x, double[4] f) nogil:
+cdef inline np_floats cubic_interpolation(np_floats x, np_real_numeric[4] f) nogil:
```

Comment:
@jni, I think one reason we aren't seeing a big speedup is probably because this might converting uint8->float64. I was hoping it would resolve it as as a float32 but that seems like alot to expect from Cython.

Maybe the correct way is to provide an `out` parameter and explicitely specify which one we want. Though that is more `compiler work` that I would like to avoid. 

Apparently people have been using templates to avoid this kind of explicit casting due to cython limitations.